src/backend/distributed/commands/multi_copy.c

```diff
@@ -3281,6 +3285,19 @@ InitializeCopyShardState(CopyShardState *shardState,
 			continue;
 		}
 
+		if (placement->groupId == GetLocalGroupId())
+		{
+			/*
+			 * if we are copying into an intermediate file we won't use local copy.
+			 * Files are visible to all transactions so we can still use local execution, therefore
+			 * we shouldn't restrict only using connection in this case.
+			 */
+			if (!isCopyToIntermediateFile)
```

Comment:
Could we use !shouldUseLocalCopy here?

Reply:
I think we can for now. I think a nicer solution would be not to set it here at all but at a more generic place, though that requires some refactoring.

Reply:
Well actually we can't because in case of copy to intermediate file, shouldUseLocalCopy will be false.